satpy/readers/fci_l1c_fdhsi.py

```diff
@@ -257,32 +260,67 @@ def get_area_def(self, key, info=None):
         self._cache[key.resolution] = area
         return area
 
-    def calibrate(self, data, key, measured, root):
-        """Calibrate data."""
+    def calibrate_to_radiances(self, data, key):
+        """Calibrate counts to radiances."""
+        radiance_units = data.attrs["units"]
+        if key.name == 'ir_38':
+            data = xr.where(((2 ** 12 - 1 < data) & (data <= 2 ** 13 - 1)),
+                            (data * data.attrs.get("warm_scale_factor", 1) +
+                             data.attrs.get("warm_add_offset", 0)),
+                            (data * data.attrs.get("scale_factor", 1) +
+                             data.attrs.get("add_offset", 0))
+                            )
+        else:
+            data = (data * data.attrs.get("scale_factor", 1) +
+                    data.attrs.get("add_offset", 0))
+
+        data.attrs["units"] = radiance_units
+
+        return data
+
+    def calibrate_counts(self, data, key):
+        """Calibrate counts to radiances, brightness temperatures, or reflectances."""
+        # counts to radiance scaling
+
+        data = self.calibrate_to_radiances(data, key)
+
         if key.calibration == 'brightness_temperature':
-            data = self._ir_calibrate(data, measured, root)
+            data = self._ir_calibrate(data, key)
         elif key.calibration == 'reflectance':
-            data = self._vis_calibrate(data, measured)
+            data = self._vis_calibrate(data, key)
+
+        return data
+
+    def calibrate(self, data, key):
+        """Calibrate data."""
+        if key.calibration == "counts":
+            # from package description, this just means not applying add_offset
+            # and scale_factor
+            data.attrs["units"] = "1"
+        elif key.calibration in ['brightness_temperature', 'reflectance', 'radiance']:
+            data = self.calibrate_counts(data, key)
         else:
-            raise ValueError(
-                    "Received unknown calibration key.  Expected "
-                    "'brightness_temperature' or 'reflectance', got "
-                    + key.calibration)
+            logger.error(
+                "Received unknown calibration key.  Expected "
+                "'brightness_temperature', 'reflectance' or 'radiance', got "
+                + key.calibration + ".")
```

Comment:
Ok, I understand the thinking here, but I'm not sure this belongs to the reader. I think it should be done more centrally so that all readers would log the error in the same way. It's not a big deal and can stay here for now, but we should consider moving this in the future.

Reply:
I agree. Leaving it for now until we don't have that functionality higher in the code.